src/geometric_algebra/nursery/graded.lean

```diff
@@ -25,21 +27,26 @@ attribute [instance] graded_module_components.b
 attribute [instance] graded_module_components.c
 
 namespace graded_module_components
-  variables {A : ℤ → Type u}
+  variables {ι : Type u} [has_zero ι] {A : ι → Type u}
+
+  def simp_grade {r s : ι} (h: r = s) : A r → A s := by {subst h, exact id}
+
   /-- objects are coercible only if they have the same grade-/
-  instance has_coe (r s : ℕ) (h: r = s) : has_coe (A r) (A s) := { coe := by {subst h, exact id}}
+  instance has_coe {r s : ι} (h: r = s) : has_coe (A r) (A s) := { coe := simp_grade h }
 end graded_module_components
 
 -- needed for the definition of `select`
 attribute [instance] dfinsupp.to_semimodule
 
 /-- Grade selection maps from objects in G to a finite set of components of substituent grades -/
 class has_grade_select
-  (A : ℤ → Type u) (G: Type u)
+  {ι : Type u} [has_zero ι]
+  (A : ι → Type u) (G: Type u)
   [graded_module_components A]
   [add_comm_group G]
+  [ring (A 0)]
```

Comment:
Without this line, `module (A 0) G` below would fail, even though this should be implied by `graded_module_components.zc`?